src/openrct2-ui/windows/TrackDesignManage.cpp

```diff
@@ -125,41 +121,34 @@ static void WindowTrackManageClose(rct_window* w)
     }
 }
 
-/**
- *
- *  rct2: 0x006D3523
- */
-static void WindowTrackManageMouseup(rct_window* w, WidgetIndex widgetIndex)
+void TrackDesignManageWindow::OnMouseUp(WidgetIndex widgetIndex)
 {
     switch (widgetIndex)
     {
         case WIDX_CLOSE:
             window_close_by_class(WindowClass::TrackDeletePrompt);
-            window_close(*w);
+            Close();
             break;
         case WIDX_RENAME:
             WindowTextInputRawOpen(
-                w, widgetIndex, STR_TRACK_DESIGN_RENAME_TITLE, STR_TRACK_DESIGN_RENAME_DESC, {},
+                this, widgetIndex, STR_TRACK_DESIGN_RENAME_TITLE, STR_TRACK_DESIGN_RENAME_DESC, {},
                 _trackDesignFileReference->name.c_str(), TrackDesignNameMaxLength);
             break;
         case WIDX_DELETE:
-            WindowTrackDeletePromptOpen();
+            WindowTrackDeletePromptOpen(_trackDesignFileReference);
             break;
     }
 }
 
-/**
- *
- *  rct2: 0x006D3523
- */
-static void WindowTrackManageTextinput(rct_window* w, WidgetIndex widgetIndex, char* text)
+void TrackDesignManageWindow::OnTextInput(WidgetIndex widgetIndex, std::string_view text)
 {
-    if (widgetIndex != WIDX_RENAME || str_is_null_or_empty(text))
+    if (widgetIndex != WIDX_RENAME || text.empty())
     {
         return;
     }
 
-    if (str_is_null_or_empty(text))
```

Comment:
This part didn't make a lot of sense to me, hence why I added a TODO

Reply:
It was added in https://github.com/OpenRCT2/OpenRCT2/commit/e666028e0274be0c3c6195e691c96ffcfb897cd5 , most likely to add a specific error message for this case. It looks like an oversight. I think you can simply get rid of the second case on line 157/145.